apis/database/v1beta1/rdsinstance_types.go

```diff
@@ -368,6 +389,10 @@ type RDSInstanceParameters struct {
 	// +optional
 	EngineVersion *string `json:"engineVersion,omitempty"`
 
+	// FromBackup specifies the details of the backup to restore when creating a new RDS instance.
+	// +optional
+	FromBackup *RestoreBackupConfiguration `json:"fromBackup"`
```

Comment:
I think this name should take into consideration what the name/structure for restoring from DB snapshots should be too. If the structure will be specific to S3, maybe S3 should be part of the name?

`RestoreDBInstanceFromS3`
`RestoreDBInstanceFromDBSnapshot`
`RestoreDBInstanceToPointInTime`

Reply:
(Disclaimer: I'm not familiar with RDS restores apart from a cursory glance at the API).

If there are in fact multiple different sources you can restore from (e.g. snapshot, S3) perhaps using a type distinction here would make sense. This is a pattern we frequently use in our APIs, e.g.:

```yaml
spec:
  forProvider:
    fromBackUp:
      source: S3
      s3:
        s3bucketName: ...
```

I don't love that we're having to do so much interpretation of the APIs. Ideally our Crossplane managed resource would map directly to one API type (e.g. `CreateDBInstance`), while here we're decorating that API type with another. Given how imperative this call is and that it in fact results in the creation of a DB instance (AFAICT) I can't think of any great alternatives though.

Reply:
Thanks, this feedback was exactly what I was looking for - I am not familiar enough with Crossplane's API patterns to understand how you guys would model a subclass/discriminator case like this. I'll update the model. Yeah, I don't really see any other options either given that the restore operation is a separate API call - it would be nicer if we could pass the restore parameters to the `CreateDBInstance` call, but it doesn't support that.

Reply:
Introduced type distinction in the latest version of the PR.